packages/utilities/src/extendComponent.ts

```diff
@@ -0,0 +1,13 @@
+import { appendFunction } from './appendFunction';
+
+/**
+ * Extends a component's lifetime methods by appending new functions to the existing lifetime functions.
+ */
+export function extendComponent<T>(parent: T, methods: { [key in keyof T]?: T[key] }): void {
```

Comment:
Might be worth doing `T extends React.Component` since the function's name is specifying component context. This should add type safety to `key` values too.

Reply:
good idea